test/python/compiler/test_transpiler.py

```diff
@@ -536,9 +536,9 @@ def test_optimize_to_nothing(self):
         circ.cx(qr[0], qr[1])
 
         after = transpile(circ, coupling_map=[[0, 1], [1, 0]],
-                          basis_gates=['u3', 'cx'])
+                          basis_gates=['u3', 'u2', 'u1', 'cx'])
```

Comment:
Is this needed because `_circuit_u3` won't simplify away a `U3Gate`, even if `theta == phi == lam == 0`? If so, this would be good to fix, either here or in a follow up PR.

Reply:
Putting this off to a new issue due to testing issues.